hudi-common/src/main/java/org/apache/hudi/common/table/read/HoodieFileGroupReader.java

```diff
@@ -346,10 +360,20 @@ public void close() throws IOException {
     }
   }
 
-  public HoodieFileGroupReaderIterator<T> getClosableIterator() {
+  public ClosableIterator<T> getClosableIterator() {
     return new HoodieFileGroupReaderIterator<>(this);
   }
 
+  /**
+   * @return An iterator over the records that wraps the engine-specific record in a HoodieRecord.
+   */
+  public ClosableIterator<HoodieRecord<T>> getClosableHoodieRecordIterator() {
```

Comment:
Is there a test for this iterator? I don't see any usages in this patch. 
can we add some sanity tests.

Reply:
Added a test to the base class `TestHoodieFileGroupReaderBase` for this